src/ux/patchesView.ts

```diff
@@ -140,201 +138,96 @@ export const patchesTreeDataProvider: TreeDataProvider<
-                      }`
-                    : filechange.path
-                } ${dot} ${capitalizeFirstLetter(filechange.state)}`,
-                resourceUri: Uri.file(filePath),
-                command:
-                  (filechange as FilechangeWithDiff).diff ??
-                  (filechange as FilechangeWithoutDiffButDiffViewerRegardless).current
-                    ? {
-                        command: 'radicle.openDiff',
-                        title: `Open changes`,
-                        tooltip: `Show this file's changes between its \
-before-the-Patch version and its latest version committed in the Radicle Patch`,
-                        arguments: [
-                          Uri.file(
-                            (filechange as Partial<FileChangeWithOldAndNew>).old?.oid ||
-                              (filechange as FilechangeWithoutDiffButDiffViewerRegardless)
-                                .current
-                              ? oldVersionUrl
-                              : emptyFileUrl,
-                          ),
-                          Uri.file(
-                            (filechange as Partial<FileChangeWithOldAndNew>).new?.oid ||
-                              (filechange as FilechangeWithoutDiffButDiffViewerRegardless)
-                                .current
-                              ? newVersionUrl
-                              : emptyFileUrl,
-                          ),
-                          `${filename} (${shortenHash(oldVersionCommitSha)} ⟷ ${shortenHash(
-                            newVersionCommitSha,
-                          )}) ${capitalizeFirstLetter(filechange.state)}`,
-                          { preview: true } satisfies TextDocumentShowOptions,
-                        ],
-                      }
-                    : undefined,
-              }
-
-              return filechangeTreeItem
-            },
-          }
-
-          return node
-        })
-        .sort((n1, n2) => (n1.relativeInRepoUrl < n2.relativeInRepoUrl ? -1 : 0))
-
-      return filechangeNodes.length
-        ? filechangeNodes
-        : [
-            `No changes between latest revision's base "${shortenHash(
-              latestRevision.base,
-            )}" and head "${shortenHash(latestRevision.oid)}" commits.`,
-          ]
+    const repoRoot = getRepoRoot()
+    if (repoRoot === undefined) {
+      throw new Error(`Failed to determine Git repository root.`)
     }
 
-    return undefined
+    const repo = gitExtensionApi.getRepository(Uri.file(repoRoot))
+    if (repo === null) {
+      throw new Error(`Failed access Git repository.`)
+    }
+
+    const changes: Change[] = await repo?.diffBetween(range.old, range.new)
+    if (changes.length === 0) {
+      return [
+        `No changes between latest revision's base "${shortenHash(
+          latestRevision.base,
+        )}" and head "${shortenHash(latestRevision.oid)}" commits.`,
+      ]
+    }
+
+    return changes
```

Comment:
In this repo we're avoiding any non-trivial return statement. All such cases should be assignment then return.